sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -809,22 +809,27 @@ class Analyzer(
           .getOrElse(i)
 
       case desc @ DescribeTable(u: UnresolvedV2Relation, _) =>
-        CatalogV2Util.loadRelation(u.catalog, u.tableName)
-            .map(rel => desc.copy(table = rel))
-            .getOrElse(desc)
+        CatalogV2Util
+          .loadRelation(u.catalog, catalogManager.catalogIdentifier(u.catalog), u.tableName)
```

Comment:
Would it be easier to have a separate util method for `UnresolvedV2Relation`? The interesting thing about `UnresolvedV2Relation` is that you have access to the `originalNameParts`. If that doesn't contain the catalog identifier, then we shouldn't need to add it to the `DataSourceV2Relation`

Reply:
Sure, I will do the refactoring.

However in terms of using `originalNameParts`, I think the major reason why we are adding the two additional fields to DSV2 relation is that both CatalogPlugin and Table implementation doesn't include the resolved unique identifier for the table. Even though the originalNameParts don't have a catalog name in it, it might point to some default catalog in which case we would still want it to be in the resolved unique identifier. Otherwise, we might have two V2Relations pointing to the same table but has different identifiers which would be inconsistent and pretty confusing.

Reply:
I think it's fine to reproduce the full identifier that was actually loaded by adding in catalog and namespace if they are missing. But I'm not sure it is a good idea to rely on this behavior since there is no guarantee.